task/servicetest/servicetest.go

```diff
@@ -55,7 +55,7 @@ func TestTaskService(t *testing.T, fn BackendComponentFactory) {
 		// but more importantly, it should exercise concurrency to catch data races.
 
 		t.Run("Task CRUD", func(t *testing.T) {
-			t.Parallel()
+			// t.Parallel()
```

Comment:
Uncomment?

Reply:
Your right. We should keep this parallel. The problem is we are using the same "system" for all the tests when we should generate a new system for each test.

Reply:
I think the reason for using the same system, and t.Parallel, was to exercise that the task systems work correctly under concurrent operation. I would prefer to rearrange the test setup over duplicating systems.

Reply:
We wrote a test just for the purpose of testing concurrency. `testTaskConcurrency` It is entirely unnecessary to use the crud test to check the concurrency as well.

Reply:
`testTaskConcurrency` should be renamed, as it doesn't exercise all the  methods covered in the other set of tests.